pkg/controllers/workspace_controller.go

```diff
@@ -703,15 +703,33 @@ func (c *WorkspaceReconciler) applyTuning(ctx context.Context, wObj *kaitov1alph
 
 			tuningParam := model.GetTuningParameters()
 			existingObj := &batchv1.Job{}
+			revisionStr := wObj.Annotations[kaitov1alpha1.WorkspaceRevisionAnnotation]
 			if err = resources.GetResource(ctx, wObj.Name, wObj.Namespace, c.Client, existingObj); err == nil {
 				klog.InfoS("A tuning workload already exists for workspace", "workspace", klog.KObj(wObj))
+
+				if existingObj.Annotations[kaitov1alpha1.WorkspaceRevisionAnnotation] != revisionStr {
+					deletePolicy := metav1.DeletePropagationForeground
+					if err := c.Delete(ctx, existingObj, &client.DeleteOptions{
+						PropagationPolicy: &deletePolicy,
+					}); err != nil {
+						return
+					}
+
+					var workloadObj client.Object
+					workloadObj, err = tuning.CreatePresetTuning(ctx, wObj, revisionStr, tuningParam, c.Client)
```

Comment:
Making sure does a new `revisionStr` need to be generated here, or same `revisionStr` as the workspace deleted?

Reply:
resolved offline